### DISCUSSION TITLE
Celebrating with a Dialogic 2.0 devlog

### DISCUSSION THREAD

**Discussion by Jowan-Spooner** [dialogic-godot/dialogic#1000] [Show and tell]
***This is the #1000 entry on this repository! We've come a long, long way.***
To celebrate this, I wanted to give a quick rundown, of what has happened with dialogic 2.0 in the past month. 

Here are the coolest new features, you should get hyped for! 
*Keep in mind that 2.0 is still in alpha and everything (especially the UI) is going to change.*

## 1. Text-based-editing
The resource-based approach for events has allowed to introduce text-based editing of events alongside the traditional timeline editor.
Here is an example of a timeline as text and in editor:
![](https://media.discordapp.net/attachments/535817927186710530/996347362358329395/DialogicPromo1.png)

## 2. Resource based timelines and characters
Timelines and characters are now resources that are stored as real files. This means you can organize them however you want in your file system.

It also allows for a much easier creation of timelines through code. Here is a short example of building a timeline in code:
```
func _ready():
        var timl = DialogicTimeline.new()
	
	# creating a new character
	var chara = DialogicCharacter.new()
	chara.name = "Cyborg13"
	chara.portraits["happy"] = {'path':"res://icon.png", 'scale':3}
	
	# creating new events
	var join = DialogicCharacterEvent.new()
	join.Character = chara
	join.Portrait = "happy"
	
	var txt = DialogicTextEvent.new()
	txt.Text = "Hi, I'm Emilio"
	txt.Character = chara
	
	timl._events = [join, txt]
	
	Dialogic.start_timeline(timl)
```

## 3. Complex conditions, Elif and Else
The new approach to the condition event allows for elif and else events. Also you can now use the properties of Autoloads in the conditions and even call some methods.
![](https://media.discordapp.net/attachments/535817927186710530/996348008344072203/DialogicPromo2.png)

## 4. Custom events, subsystems and settings
After dialogic was initally released, mainly to be used for visual novels, it turned out people wanted to do a lot of things with it, that it wasn't originally intended for. We want to embrace this, and have thus worked very hard to make dialogic as easy to extend as possible. 
You can very easily create 
- new events
- subsystems (scripts that hold functionality related to one feature) 
- settings editors to dialogic

This means that it's very easy to create and distribute complete systems build for dialogic. I could imagine an inventory system, a quest system or systems that are very specific to a game. If everything works out theses systems should blend in with dialogic as if they really belong there.

At the same time, you can also remove all the built-in events/systems in case you don't want them.

## 5. New way of themeing
Dialogic 2.0 has a very different approach to styling your dialog. The dialog parser and display functionality have been completely split up. There is no theme editor anymore. 

Here is how it works now:
- Dialogic provides so called DisplayNodes. Most of these nodes are basically normal nodes like Label, Control, RichTextLabel, Button, etc. Some of them have some default functionality built-in (like the DialogText scene that can animate it's text like we are used to). 

- You can instance these display scenes in your scene wherever you want. You can even have multiple NameLabels or provide the same choices in two seperate places. No clue why you would want that, but who knows.

- If you want different themes, than you can use the Theme node. It's just a control that allows you to enter a theme name. It will be hidden/shown based on the active theme. This means you can add all the elements you want to vary in a theme as children to that node. 

This all means that you can use ALL of godots themeing capabilities and be MUCH more flexible with the layout. Dialogic 2.0 will likely come with multiple presets of Display-Node scenes, but it should be as easy as never before to customize your game.

## 6. Translation support
You can add translations to your timelines. It uses the godot built-in translation system, meaning you will have to edit the translations in a csv editor. Because events have unique translation id's you can still edit your timelines to a certain degree, after translations where enabled. We still recommend adding them in later stages of production though.

---

Hopefully this has made you excited for the future of dialogic. We are steadily working on 2.0 and you can track the progress [here](https://github.com/coppolaemilio/dialogic/issues?q=is%3Aopen+is%3Aissue+milestone%3A%22Version+2.0%22).
I find it likely that there will be a first beta some time in august, but won't make any promises.

Until then, have fun creating!

**Comment by coppolaemilio** [comment]
I should make a video covering the progress so far! Thank you so much for the rundown, it is great to see it all coming together 👏🏻

**Reply by zaknafean** [reply]
I can't wait to use the real file system for this stuff! Literally my biggest pain point with Dialogic and a large project is searching for timelines and creating new ones taking forever. Especially bad when I need a dozen short ones.
